internal/pkg/log/hooks.go

```diff
@@ -0,0 +1,110 @@
+package log
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type FileNameHook struct {
+	field      string
+	skipPrefix []string
+	formatter  logrus.Formatter
+	Formatter  func(file, function string, line int) string
+}
+
+type wrapper struct {
+	old  logrus.Formatter
+	hook *FileNameHook
+}
+
+// NewFilenameHook creates a new default FileNameHook
+func NewFilenameHook() *FileNameHook {
+	return &FileNameHook{
+		field:      "file",
+		skipPrefix: []string{"log/", "logrus/", "logrus@"},
+		Formatter: func(file, function string, line int) string {
+			return fmt.Sprintf("%s:%d", file, line)
+		},
+	}
+}
+
+// Levels returns the levels for which the hook is activated. This contains
+// currently only the DebugLevel
+func (f *FileNameHook) Levels() []logrus.Level {
+	return []logrus.Level{logrus.DebugLevel}
+}
+
+// Fire executes the hook for every logrus entry
+func (f *FileNameHook) Fire(entry *logrus.Entry) error {
+	if f.formatter != entry.Logger.Formatter {
+		f.formatter = &wrapper{entry.Logger.Formatter, f}
+	}
+	entry.Logger.Formatter = f.formatter
+	return nil
+}
+
+// Format returns the log format including the caller as field
+func (w *wrapper) Format(entry *logrus.Entry) ([]byte, error) {
+	field := entry.WithField(
+		w.hook.field,
+		w.hook.Formatter(w.hook.findCaller()),
+	)
+	field.Level = entry.Level
+	field.Message = entry.Message
+	return w.old.Format(field)
+}
+
+// findCaller returns the file, function and line number for the current call
+func (f *FileNameHook) findCaller() (file, function string, line int) {
+	var pc uintptr
+	for i := 0; i < 10; i++ {
+		pc, file, line = caller(5 + i)
```

Comment:
OK, I'll bite, what's the significance of 10 and 5?  A comment and/or variables with names that clarify would be appreciated.

Reply:
Yes I see. I added const variables and a corresponding description to it.